### PR TITLE
Use middleware on feature endpoints to check cluster and organization

### LINKED ISSUE #2195: Do not allow feature actions in case of wrong organization
**Describe the bug**
I have a GKE cluster with Auto DNS feature. I want to list the feature details (`/pipeline/api/v1/orgs/2/clusters/{clusterID}/features/dns`) i got the right response. After i change the organization (from `colin014`(2) to `banzaicloud`(10)) i still got the right response but the cluser isn't belong to `banzaicloud` organization.

I can do any action (like active & deactivate) on features of cluster

**Expected behavior**
Do not allow any action with feature on clusters which not belongs to the organization. Check the organization with the organization of the cluster.

### PR DESCRIPTION
**Pull request by colin014**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #2195 
| License         | Apache 2.0

### What's in this PR?
Use middleware on feature endpoints to check cluster and organization. In case of wrong cluster/organization pair the new response:
```json
{
    "type": "about:blank",
    "title": "Not Found",
    "status": 404,
    "detail": "could not get cluster from database: could not find cluster by ID: cluster not found"
}
```

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)

### CODE REVIEW

File: cmd/pipeline/main.go [hunk 1]
```diff
@@ -563,16 +564,7 @@ func main() {
...
-				// TODO: move this to a middleware
-				router.Use(func(c *gin.Context) {
-					clusterID, ok := ginutils.UintParam(c, "id")
-					if !ok {
-						return
-					}
-
-					c.Request = c.Request.WithContext(ctxutil.WithClusterID(c.Request.Context(), clusterID))
-				})
+				router.Use(features.NewMiddleware(clusterManager, errorHandler))
```
Comment (sagikazarmark) [inline cmd/pipeline/main.go]: Use this middleware on every cluster endpoint

File: internal/clusterfeature/features/middleware.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		}
+
+		orgID, ok := ginutils.UintParam(c, "orgid")
+		if !ok {
+			return
+		}
+
+		_, err := manager.GetClusterByID(c, orgID, clusterID)
+		if err != nil {
+			errorHandler.Handle(err)
```
Comment (sagikazarmark) [inline internal/clusterfeature/features/middleware.go]: Only pass the error to the error handler, if it's not a not found error.

File: internal/clusterfeature/features/middleware.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+			problem := problems.NewDetailedProblem(http.StatusNotFound, err.Error())
```
Comment (sagikazarmark) [inline internal/clusterfeature/features/middleware.go]: The manager might respond with error for various reasons (eg. DB unavailable). Check if the returned error is a not found error (the root cause implements`interface{NotFound() bool}` and it returns true, see other examples in the repo) and return 404 only if that's the case.

**Comment by colin014** [comment]
thx @sagikazarmark i made the requested changes

**Pull request merged**